cmd/zpool/zpool_main.c

```diff
@@ -1009,8 +1009,9 @@ add_prop_list_default(const char *propname, const char *propval,
 }
 
 /*
- * zpool add [-fgLnP] [-o property=value] <pool> <vdev> ...
+ * zpool add [-afgLnP] [-o property=value] <pool> <vdev> ...
  *
+ *	-a	Disable the ashift validation checks
```

Comment:
Why add a new flag for this? Why not just expand the use of `-f`? I believe that currently `-f` effects both "in use" or "different replication level". It seems reasonable to add: or "different ashift".

Reply:
The main concern I have with using the `-f` option is that it is often used to bypass the in use check when adding devices which belonged to decommissioned pools.  As a result consumers may inadvertently bypass the ashift validation and then find that they can't perform device removal. This is why I wanted to make it more explicit.

Reply:
Yeah, I suppose that is always a risk when you have more than a single condition being covered by an "override" flag. I still think that it would be reasonable to use the `-f` flag here, as it's an administrative error to "blindly" use an override without knowing all the conditions being overridden. However, I see your point and so don't insist on a single flag :-). Go with what you think best.

Reply:
It would be good to create a unified way to specify what exactly should be overridden, like `-F ashift -F blabla`, etc.  I am not a big fan of new options, since we are pretty low on letters.

Reply:
There is an open [draft PR](https://github.com/openzfs/zfs/pull/14754) which starts adding this functionality for `zpool create`.  It'd be nice to extend whatever we settle on there to the other `zpool` commands.   I'm fine with adding another flag for now, but I think it'd be nice to move in that direction.